src/server/plugins/engine/components/NumberField.ts

```diff
@@ -23,7 +23,11 @@ export class NumberField extends FormComponent {
 
     const { options, schema, title } = def
 
-    let formSchema = joi.number().label(title.toLowerCase()).required()
+    let formSchema = joi
+      .number()
+      .custom(getValidatorPrecision(this))
```

Comment:
Preventing rounding using `.prefs({ convert: false })` made every "string number" invalid 😔

I've used a custom validator instead so we can keep string-to-number conversion